fusesoc/vlnv.py

```diff
@@ -167,16 +167,10 @@ def vln_str(self):
         return f"{self.vendor}:{self.library}:{self.name}"
 
 
-def compare_relation(vlvn_a, relation, vlvn_b):
+def compare_relation(vlvn_a: Vlnv, relation: str, vlvn_b: Vlnv):
     """Compare two VLVNs with the provided relation. Returns boolan."""
     from okonomiyaki.versions import EnpkgVersion
 
-    if (
-        not isinstance(vlvn_a, Vlnv)
-        or not isinstance(relation, str)
-        or not isinstance(vlvn_b, Vlnv)
-    ):
-        return False
```

Comment:
I really like this change, but would it be wise to have static type checking (e.g. mypy) or runtime validation (e.g. `pydantic.validate_call`) before removing checks like this?

Reply:
Ah, because there is nothing in Python that actually enforces the types by default, right? Yes, let's introduce type checking somehow, but I think this change can still go in before that. YOLO!